src/views/datalist/data-list-directive.js

```diff
@@ -186,25 +194,104 @@
             city: "New York",
             state: "New York"
           },
-        ]
+        ];
+
+        var performAction = function (action, item) {
+          $scope.eventText = item.name + " : " + action.name + "\n" + $scope.eventText;
+        };
+
+        $scope.actions = [
+          {
+            name: 'Action',
+            title: 'Perform an action',
+            actionFn: performAction
+          },
+          {
+            name: 'Another Action',
+            title: 'Do something else',
+            actionFn: performAction
+          },
+          {
+            name: 'Disabled Action',
+            title: 'Unavailable action',
+            actionFn: performAction,
+            isDisabled: true
+          },
+          {
+            name: 'Something Else',
+            title: '',
+            actionFn: performAction
+          },
+          {
+            isSeparator: true
+          },
+          {
+            name: 'Grouped Action 1',
+            title: 'Do something',
+            actionFn: performAction
+          },
+          {
+            name: 'Grouped Action 2',
+            title: 'Do something similar',
+            actionFn: performAction
+          }
+        ];
       }
     ]);
   </file>
 </example>
  */
-angular.module('patternfly.views').directive('pfDataList', function (pfUtils) {
+angular.module('patternfly.views').directive('pfDataList',
+  function ($compile, $timeout, $window, pfUtils) {
     'use strict';
     return {
       restrict: 'A',
       scope: {
         config: '=?',
         items: '=',
-        eventId: '@id'
+        actions: '=?',
+        updateActionForItemFn: '=?'
       },
       transclude: true,
       templateUrl: 'views/datalist/data-list.html',
       controller: ['$scope',
         function ($scope) {
+          var setDropMenuLocation = function (parentDiv) {
+            var dropButton = parentDiv.querySelector('.dropdown-toggle');
+            var dropMenu =  parentDiv.querySelector('.dropdown-menu');
+            var buttonRect = dropButton.getBoundingClientRect();
+            var menuRect = dropMenu.getBoundingClientRect();
+            var top = buttonRect.top + buttonRect.height;
+            var left = buttonRect.left + buttonRect.width - menuRect.width;
+            var docHeight = $window.innerHeight;
+
+            if (top + menuRect.height > docHeight) {
+              top = docHeight - menuRect.height;
+            }
+
+            dropMenu.style.top = top + "px";
+            dropMenu.style.left = left + "px";
```

Comment:
based on where you are on the page you may need to set dropMenu.style.bottom instead